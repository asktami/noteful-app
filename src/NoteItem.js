import React, { useContext } from 'react';
import { NavLink } from 'react-router-dom';

import config from './config';
import NotefulContext from './NotefulContext';

// this function 1st deletes via the API, then from state
// context.deleteNote = the updater function, to update state in context
function deleteNoteRequest(noteId, cb) {
	fetch(config.API_NOTES + `/${noteId}`, {
		method: 'DELETE',
		headers: {
			'content-type': 'application/json'
		}
	})
		.then(res => {
			if (!res.ok) {
				// get the error message from the response,
				return res.json().then(error => {
					// then throw it
					throw error;
				});
			}
			return res.json();
		})
		.then(data => {
			// call the callback function when the request is successful
			// this is where the App component can remove it from state
			cb(noteId);
		})
		.catch(error => {
			console.error('delete note errorNotes = ', error);
			this.setState({ errorNotes: error });
		});
}

const NoteItem = props => {
	console.log('props', JSON.stringify(props));
	return (
		<NotefulContext.Consumer>
			{/*
			Use the Consumer to grab values from contex

 			QUESTION: what is context?
			ANSWER:
			is it equal to the object inside NotefulCcontext.js?
			an anonymous function with the parameter context automatically defined by {context => ... }, don't know where the parameter function comes from, don't need to know where it comes from because we're using it to render a result; context is a variable containing data; this function is called by line 39 NotefulContext.Consumer

			NOTE: context could be any word since its just the parameter label
			*/}

			{context => (
				<div className="note-item">
					{/*
			THIS CAUSED A staticContent ERROR:
			<NavLink to={`/note/${note.id}`} {...props}>
				<h3>{note.title}</h3>
			</NavLink> */}

					<NavLink to={{ pathname: `/note/${props.id}`, props: props }}>
						<h3>{props.name}</h3>
					</NavLink>

					<div className="button-container">
						<span>
							console.log('got here'); Modified on
							<span className="note-datemod">
								if(props.modified != undefined){' '}
								{props.modified.split('T', 1)[0]}
							</span>
						</span>
						<span>
							<button
								className="btn-delete"
								onClick={() => {
									deleteNoteRequest(props.id, context.deleteNote);
								}}
							>
								-
							</button>
						</span>
					</div>
				</div>
			)}
		</NotefulContext.Consumer>
	);
};
export default NoteItem;
