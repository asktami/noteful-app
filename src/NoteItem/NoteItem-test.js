import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NoteItem from './NoteItem';

// hardcode modified date so that test will pass when deploy to server in a different timezone

describe(`NoteItem component`, () => {
	const props = {
		note: {
			id: 1,
			id_folder: 101,
			title: 'test-note-title',
			modified: new Date(Date.UTC(2021, 3, 11, 3, 4, 5)).toLocaleString('en-US', { timeZone: 'America/New_York' }),
		},
	};

	it('renders a NoteItem by default', () => {
		const wrapper = shallow(<NoteItem {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the NoteItem given props', () => {
		const wrapper = shallow(<NoteItem {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
