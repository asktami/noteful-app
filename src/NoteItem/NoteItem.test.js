import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NoteItem from './NoteItem';

describe('Dummy NoteItem test', () => {
	test('adding 1 + 2 should return 3', () => {
		expect(2 * 2).toBe(4);
	});
});
