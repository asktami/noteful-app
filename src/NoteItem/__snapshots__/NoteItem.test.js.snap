// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem component renders a NoteItem by default 1`] = `
<div
  className="note-item"
>
  <NavLink
    to={
      Object {
        "pathname": "/notes/1",
        "props": Object {
          "note": Object {
            "id": 1,
            "id_folder": 101,
            "modified": 2021-04-11T03:04:05.000Z,
            "title": "test-note-title",
          },
        },
      }
    }
  >
    <h3 />
  </NavLink>
  <div
    className="button-container"
  >
    <span>
      Modified on
       
      <span
        className="note-datemod"
      >
         
        Sat Apr 10 2021 23:04:05 GM
      </span>
    </span>
    <span>
      <NavLink
        to="/edit-note/1"
      >
        <button
          className="btn-edit-note"
        >
          ✎
        </button>
      </NavLink>
         
      <button
        className="btn-delete-note"
        onClick={[Function]}
      >
        -
      </button>
    </span>
  </div>
</div>
`;

exports[`NoteItem component renders the NoteItem given props 1`] = `
<div
  className="note-item"
>
  <NavLink
    to={
      Object {
        "pathname": "/notes/1",
        "props": Object {
          "note": Object {
            "id": 1,
            "id_folder": 101,
            "modified": 2021-04-11T03:04:05.000Z,
            "title": "test-note-title",
          },
        },
      }
    }
  >
    <h3 />
  </NavLink>
  <div
    className="button-container"
  >
    <span>
      Modified on
       
      <span
        className="note-datemod"
      >
         
        Sat Apr 10 2021 23:04:05 GM
      </span>
    </span>
    <span>
      <NavLink
        to="/edit-note/1"
      >
        <button
          className="btn-edit-note"
        >
          ✎
        </button>
      </NavLink>
         
      <button
        className="btn-delete-note"
        onClick={[Function]}
      >
        -
      </button>
    </span>
  </div>
</div>
`;
